src/web/demo.rkt

```diff
@@ -328,19 +328,58 @@
        (when (eof-object? formula)
          (raise-herbie-error "no formula specified"))
        (parse-test formula)
-       (define hash (sha1 (open-input-string formula-str)))
+       (define hash (sha1 (open-input-string (~s 'improve formula (get-seed)))))
        (body hash formula))]
     [_
      (response/error "Demo Error"
                      `(p "You didn't specify a formula (or you specified several). "
                          "Please " (a ([href ,go-back]) "go back") " and try again."))]))
 
+(define (improve-start-extract post-data)
+  (define form-str (hash-ref post-data 'formula))
+  (define form-seed (hash-ref post-data 'seed))
```

Comment:
Won't this throw if the `formula` or `seed` fields aren't present?

Reply:
Thank you. Is any Racket function with `[failure-result]` option have the chance of throwing an error?

I started some sense of error handling and basic test for this.